src/TableFunctions/Hive/TableFunctionHive.cpp

```diff
@@ -0,0 +1,90 @@
+#include <TableFunctions/Hive/TableFunctionHive.h>
+#if USE_HIVE
+#include <memory>
+#include <type_traits>
+#include <Common/Exception.h>
+#include <Common/ErrorCodes.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ParserPartition.h>
+#include <Parsers/ExpressionListParsers.h>
+#include <Parsers/parseQuery.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/evaluateConstantExpression.h>
+#include <Storages/Hive/HiveSettings.h>
+#include <Storages/Hive/StorageHive.h>
+#include <TableFunctions/TableFunctionFactory.h>
+#include <TableFunctions/parseColumnsListForTableFunction.h>
+#include <base/logger_useful.h>
+
+namespace DB
+{
+    namespace ErrorCodes
+    {
+        extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    }
+
+    void TableFunctionHive::parseArguments(const ASTPtr & ast_function_, ContextPtr context_)
+    {
+        ASTs & args_func = ast_function_->children;
+        if (args_func.size() != 1)
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Table function '{}' must have arguments.", getName());
+
+        ASTs & args = args_func.at(0)->children;
+
+        const auto message = fmt::format(
+            "The signature of function {} is:\n"
+            " - hive_url, hive_database, hive_table, structure, partition_by_keys",
+            getName());
+
+        if (args.size() != 5)
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, message);
+
+        for (auto & arg : args)
+            arg = evaluateConstantExpressionOrIdentifierAsLiteral(arg, context_);
+
+        hive_metastore_url = args[1]->as<ASTLiteral &>().value.safeGet<String>();
+        hive_database = args[2]->as<ASTLiteral &>().value.safeGet<String>();
+        hive_table = args[3]->as<ASTLiteral &>().value.safeGet<String>();
+        table_structure = args[4]->as<ASTLiteral &>().value.safeGet<String>();
+        partition_by_def = args[5]->as<ASTLiteral &>().value.safeGet<String>();
```

Comment:
args.size() is 5, why don't you start from zero index?

Reply:
This is my fault.